src/neo4j_graphrag/experimental/pipeline/kg_builder.py

```diff
@@ -0,0 +1,221 @@
+        potential_schema (Optional[List[tuple]]): A list of potential schema relationships.
+        from_pdf (bool): Determines whether to include the PdfLoader in the pipeline.
+                         If True, expects `file_path` input in `run` methods.
+                         If False, expects `text` input in `run` methods.
+        text_splitter (Optional[Any]): A text splitter component. Defaults to FixedSizeSplitter().
+        pdf_loader (Optional[Any]): A PDF loader component. Defaults to PdfLoader().
+        kg_writer (Optional[Any]): A knowledge graph writer component. Defaults to Neo4jWriter().
+        on_error (OnError): Error handling strategy. Defaults to OnError.RAISE.
+    """
+
+    def __init__(
+        self,
+        llm: LLMInterface,
+        driver: neo4j.Driver,
+        entities: Optional[List[str]] = None,
+        relations: Optional[List[str]] = None,
+        potential_schema: Optional[List[tuple[str, str, str]]] = None,
+        from_pdf: bool = True,
+        text_splitter: Optional[Any] = None,
+        pdf_loader: Optional[Any] = None,
+        kg_writer: Optional[Any] = None,
+        on_error: OnError = OnError.RAISE,
+    ):
+        self.entities = [SchemaEntity(label=label) for label in entities or []]
+        self.relations = [SchemaRelation(label=label) for label in relations or []]
+        self.potential_schema = potential_schema if potential_schema is not None else []
+
+        self.from_pdf = from_pdf
+        self.llm = llm
+        self.driver = driver
+        self.text_splitter = text_splitter or FixedSizeSplitter()
+        self.on_error = on_error
+        self.pdf_loader = pdf_loader if pdf_loader is not None else PdfLoader()
+        self.kg_writer = kg_writer if kg_writer is not None else Neo4jWriter(driver)
+
+        self.pipeline = self._build_pipeline()
+
+    def _build_pipeline(self) -> Pipeline:
+        pipe = Pipeline()
+
+        pipe.add_component(self.text_splitter, "splitter")
+        pipe.add_component(SchemaBuilder(), "schema")
+        pipe.add_component(
+            LLMEntityRelationExtractor(llm=self.llm, on_error=self.on_error),
+            "extractor",
+        )
+        pipe.add_component(self.kg_writer, "writer")
+
+        if self.from_pdf:
+            pipe.add_component(self.pdf_loader, "loader")
+            pipe.connect(
+                "loader",
+                "splitter",
+                input_config={"text": "loader.text"},
+            )
+            pipe.connect(
+                "schema",
+                "extractor",
+                input_config={
+                    "schema": "schema",
+                    "document_info": "loader.document_info",
+                },
+            )
+        else:
+            pipe.connect(
+                "schema",
+                "extractor",
+                input_config={
+                    "schema": "schema",
+                    "document_info.path": "direct_text_input",
```

Comment:
This syntax is not possible at the moment, this is a limitation of how the pipeline understands the parameters. We have to use:
```
  "document_info": {
      "path": "direct_text_input",
  }
```

Reply:
https://github.com/neo4j/neo4j-graphrag-python/pull/165/commits/eaad20aaecbe132b58cd93cd1e4822ab014237ff